tephi/artists.py

```diff
@@ -0,0 +1,307 @@
+import matplotlib.artist
+import numpy as np
+from scipy.interpolate import interp1d
+from shapely.geometry import LineString, Polygon
+from shapely.prepared import prep
+
+from .constants import default
+from .isopleths import Isobar, WetAdiabat, HumidityMixingRatio
+from .transforms import convert_xy2Tt, convert_Tt2pT
+
+
+class IsoplethArtist(matplotlib.artist.Artist):
+    def __init__(self):
+        super(IsoplethArtist, self).__init__()
+        self._isopleths = None
+
+    def _locator(self, x0, x1, y0, y1):
+        temperature, theta = convert_xy2Tt([x0, x0, x1, x1], [y0, y1, y1, y0])
+        bbox = prep(Polygon(zip(temperature, theta)))
+        mask = [bbox.intersects(item.geometry) for item in self._isopleths]
+        mask = np.asarray(mask)
+
+        if self.nbins:
+            indices = np.where(mask)[0]
+            if indices.size:
+                if self.nbins < indices.size:
+                    mask[:] = False
+                    upint = indices.size + self.nbins - 1
+                    # this is an ugly solution, I'm sure there must be better ones
+                    mask[indices[:: upint // self.nbins + 1]] = True
+
+        return mask
+
+
+class IsobarArtist(IsoplethArtist):
+    def __init__(
+        self,
+        ticks=None,
+        line=None,
+        text=None,
+        min_theta=None,
+        max_theta=None,
+        nbins=None,
+    ):
+        super(IsobarArtist, self).__init__()
+        if ticks is None:
+            ticks = default.get("isobar_ticks")
+        self.ticks = ticks
+        self._kwargs = {}
+        if line is None:
+            line = default.get("isobar_line")
+        self._kwargs["line"] = line
+        if text is None:
+            text = default.get("isobar_text")
+        self._kwargs["text"] = text
+        if min_theta is None:
+            min_theta = default.get("isobar_min_theta")
+        self.min_theta = min_theta
+        if max_theta is None:
+            max_theta = default.get("isobar_max_theta")
+        self.max_theta = max_theta
+        if nbins is None:
+            nbins = default.get("isobar_nbins")
+        elif nbins < 2 or isinstance(nbins, str):
+            nbins = None
+        self.nbins = nbins
+
+    @matplotlib.artist.allow_rasterization
+    def draw(
+        self, renderer, line=None, text=None, min_theta=None, max_theta=None
+    ):
```

Comment:
I'm not sure that you can use `allow_rasterization` and also pass extra keywords to the `draw` method any more.  C.f. https://github.com/SciTools/cartopy/pull/2333.